cmd/podman/common/specgen.go

```diff
@@ -655,26 +655,53 @@ func FillOutSpecGen(s *specgen.SpecGenerator, c *ContainerCLIOpts, args []string
 	return nil
 }
 
-func makeHealthCheckFromCli(inCmd, interval string, retries uint, timeout, startPeriod string) (*manifest.Schema2HealthConfig, error) {
+func makeHealthCheckFromCli(inCmd string, interval string, retries uint, timeout, startPeriod string) (*manifest.Schema2HealthConfig, error) {
+	cmdArr := &[]string{}
+	isArr := false
+	cmdSplit := strings.SplitN(inCmd, " ", 2) // default for compat
+	if strings.Contains(inCmd, "[") {         // given an array, need to unmarshall it into a pointer
+		isArr = true
+		err := json.Unmarshal([]byte(inCmd), cmdArr)
+		if err != nil {
+			return nil, err
+		}
+		cmdSplit = *cmdArr
+	}
+	concat := ""
+	if cmdSplit[0] != "CMD" && cmdSplit[0] != "none" && cmdSplit[0] != "CMD-SHELL" { // if it isnt a cmd-shell or a cmd, it is a command
+		if isArr && len(cmdSplit) > 1 { // an array of consecutive commands
+			cmdSplit = append([]string{"CMD"}, cmdSplit...)
+		} else { // one singular command
+			if len(cmdSplit) == 1 {
+				concat = cmdSplit[0]
+			} else {
+				concat = strings.Join(cmdSplit[1:], " ")
+			}
+			cmdSplit = append([]string{"CMD-SHELL"}, concat)
```

Comment:
I'm confused as to the necessity of this. We've split the command apart on whitespace, but now we're concatenating it together again? Seems unnecessary. Seems like it would be easier to just check for certain prefixes? Especially given that I don't think that `CMD-SHELL` inside a JSON array is actually valid.

Reply:
This is for the purpose of compat vs non compat. The compat side will always include `CMD` or `CMD-SHELL` while most podman users will not. This is similar to how this was done before, there are just more codepaths now, I think all compat healthchecks were broken and duplicating `CMD-SHELL` twice

Reply:
Sure, but that doesn't address the core of my concern - there's no reason I can see to split the command on whitespace, then recombine it again into a single string for the typical CLI case. It seems like you really have two separate cases - CLI input as a JSON array, and CLI input as a raw string - and both of them can be parsed differently, because I don't believe that something like `--health-cmd '["CMD-SHELL", "echo hello world"]'` is valid.

Reply:
hmmm okay I see what you are saying, this introduces an issue with docker though, since all of their commands have to have either cmd-shell or cmd when passed to this function.